packages/geo/src/lib/layer/layer-legend/layer-legend.component.ts

```diff
@@ -27,5 +28,25 @@ export class LayerLegendComponent {
   }
   private _legend;
 
-  constructor() {}
+  constructor(private capabilitiesService: CapabilitiesService) {}
+
+   computeItemTitle(item): string {
+    const layerOptions = this.layer.dataSource.options as any;
+    if (layerOptions.type === 'wms' && layerOptions.optionsFromCapabilities) {
```

Comment:
That if statement contains the whole function body. Do an early return before entering that "if" if the conditions are not met.